pandas/tests/groupby/test_groupby.py

```diff
@@ -965,10 +967,12 @@ def test_omit_nuisance_agg(df, agg_function, numeric_only):
     if agg_function in no_drop_nuisance and not numeric_only:
         # Added numeric_only as part of GH#46560; these do not drop nuisance
         # columns when numeric_only is False
-        klass = ValueError if agg_function in ("std", "sem") else TypeError
-        msg = "|".join(["[C|c]ould not convert", "can't multiply sequence"])
-        if agg_function == "median":
-            msg = r"Cannot convert \['one' 'three' 'two'\] to numeric"
+        if agg_function in ("std", "sem"):
+            klass = ValueError
+            msg = "could not convert string to float: 'one'"
```

Comment:
Is the difference intended here?

Reply:
Previously we checked multiple possibilities for the message, of which this was one. By separating out the ValueError from the TypeError cases, we only have to check the message associated with the ValuError.

At least, I think that's what you're asking about.

Reply:
Sorry, I meant that std and see have different errors

Reply:
Ah, I see! I do think there is something to improve here. If the algorithm _actually tries_ to convert the value to a numeric one and fails, that to me is a ValueError. If, on the other hand, it sees that the object is a string and then doesn't even make an attempt to convert it, that would be a TypeError.

It would be good if we made this consistent by either (a) always trying to convert or (b) never trying to convert. However, even without this inconsistency currently `mean` et al don't try to make the conversion and still raise "Could not convert":

```
df = DataFrame({'a': [1, 1, 2], 'b': list('456')})
gb = df.groupby('a')
gb.mean()
# TypeError: Could not convert string '45' to numeric
```

This should be improved, but separate from this PR.